lib/util/zmqutil.js

```diff
@@ -1,30 +1,211 @@
-//
-// Copyright © 2022 contains code contributed by Orange SA, authors: Denis Barbaron - Licensed under the Apache license 2.0
-//
-
-// ISSUE-100 (https://github.com/openstf/stf/issues/100)
-
-// In some networks TCP Connection dies if kept idle for long.
-// Setting TCP_KEEPALIVE option true, to all the zmq sockets
-// won't let it die
-
-import zmq from 'zeromq'
-import logger from './logger.js'
-const log = logger.createLogger('util:zmqutil')
-
-export const socket = function(...args) {
-    let sock = zmq.socket(...args)
-
-  ;['ZMQ_TCP_KEEPALIVE', 'ZMQ_TCP_KEEPALIVE_IDLE', 'ZMQ_IPV6'].forEach(function(opt) {
-        if (process.env[opt]) {
-            try {
-                sock.setsockopt(zmq[opt], Number(process.env[opt]))
-            }
-            catch (err) {
-                log.warn('ZeroMQ library too old, no support for %s', opt)
-            }
-        }
-    })
-
-    return sock
-}
+//
+// Copyright © 2025 contains code contributed by V Kontakte LLC - Licensed under the Apache license 2.0
+//
+// This wrapper is designed to make 0MQ v6 backwards compatible with v5
+
+import * as zmq from 'zeromq'
+import logger from './logger.js'
+import {EventEmitter} from 'events'
+const log = logger.createLogger('util:zmqutil')
+
+const socketTypeMap = {
+    pub: zmq.Publisher
+    , sub: zmq.Subscriber
+    , push: zmq.Push
+    , pull: zmq.Pull
+    , dealer: zmq.Dealer
+    , router: zmq.Router
+    , pair: zmq.Pair
+    , req: zmq.Request
+    , reply: zmq.Reply
+}
+
+export const socket = (type) => {
+    if (!Object.prototype.hasOwnProperty.call(socketTypeMap, type)) {
+        throw new Error(`Unsupported socket type: ${type}`)
+    }
+
+    const SocketClass = socketTypeMap[type]
+    const originalSocket = new SocketClass()
+
+    const emitter = new EventEmitter()
+
+    const state = {
+        endpoints: new Set()
+        , isActive: true
+        , reconnectDelay: 1000
+    }
+
+    const wrapper = {
+        type: type
+
+        , emit: emitter.emit.bind(emitter)
+        , on: emitter.on.bind(emitter)
+        , once: emitter.once.bind(emitter)
+        , removeListener: emitter.removeListener.bind(emitter)
+
+        , bindSync: originalSocket.bindSync.bind(originalSocket)
+
+        , connect: (endpoint) => {

```

Comment:
having a singleton object class that extends EventEmmiter is better for future type checking